misc/allow-block-list/src/lib.rs

```diff
@@ -65,7 +65,7 @@ use libp2p_core::{Endpoint, Multiaddr};
 use libp2p_identity::PeerId;
 use libp2p_swarm::{
     dummy, CloseConnection, ConnectionDenied, ConnectionId, FromSwarm, NetworkBehaviour,
-    NetworkBehaviourAction, PollParameters, THandler, THandlerInEvent, THandlerOutEvent,
+    PollParameters, THandler, THandlerInEvent, THandlerOutEvent, ToSwarm,
```

Comment:
With this change `libp2p-allow-block-list` depends on `libp2p-swarm` `>=0.42.1`, but its `Cargo.toml` allows `libp2p-swarm` `v0.42.0`.

I don't think this is a pressing issue, thus not sure whether to release a patch release of `libp2p-allow-block-list` with the above restriction. What do you think @thomaseizinger?

The same applies to other crates below that we have released since.

Reply:
Right, it is an issue. We should have bumped the dependency version but not the version of each dependent crate.

I'll send a fix.

Can we mitigate this with an automated check somehow? Perhaps `-Z minimal-versions` helps?

Reply:
Done here: https://github.com/libp2p/rust-libp2p/pull/3711.

Reply:
> Can we mitigate this with an automated check somehow? Perhaps `-Z minimal-versions` helps?

I think I found a solution: https://github.com/libp2p/rust-libp2p/pull/3715